pkg/controller/certificates/certificate_controller.go

```diff
@@ -41,7 +41,7 @@ type AutoApprover interface {
 }
 
 type Signer interface {
-	Sign(csr *certificates.CertificateSigningRequest) ([]byte, error)
+	Sign(csr *certificates.CertificateSigningRequest) (*certificates.CertificateSigningRequest, error)
```

Comment:
It feels a bit odd to be returning a `CertificateSigningRequest` as well. I like having the struct instead of raw bytes so we can easily evolve the API over time, but if we ever have a discrepancy between inputs and outputs, then we'll just be ignoring fields half the time. Why not create a `CertificateSigningResponse` instead so the inputs and outputs can evolve independently?